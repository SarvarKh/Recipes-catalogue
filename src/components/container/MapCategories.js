import { connect } from 'react-redux';
import PropTypes from 'prop-types';

const MapCategories = ({ categories, handleClick }) => (
  categories.map((category) => (
    <button type="button" onClick={() => handleClick(category.strCategory)} key={category.idCategory} className="cat-item">
      <div className="category">
        <img src={category.strCategoryThumb} alt="Category" />
        <div className="cat-name tooltip">
          {category.strCategory}
          <span className="category-desc tooltiptext">{category.strCategoryDescription}</span>
        </div>
      </div>
    </button>
  ))
);

const mapStateToProps = (state) => ({
  categories: state.categories.categories,
});

export default connect(mapStateToProps, null)(MapCategories);

MapCategories.propTypes = {
  categories: PropTypes.instanceOf(Array).isRequired,
};
