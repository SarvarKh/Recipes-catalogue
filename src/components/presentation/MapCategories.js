const MapCategories = ({ categories, handleClick }) => (
  categories.map((category) => (
    <button type="button" onClick={() => handleClick(category.strCategory)} key={category.idCategory} className="cat-item">
      <div className="category">
        <img src={category.strCategoryThumb} alt="Category" />
        <div className="cat-name tooltip">
          {category.strCategory}
          <span className="category-desc tooltiptext">{category.strCategoryDescription}</span>
        </div>
      </div>
    </button>
  ))
);

export default MapCategories;
