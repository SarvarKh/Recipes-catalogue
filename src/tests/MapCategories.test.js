import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import MapCategories from '../components/presentation/MapCategories';

const categories = [
    {
        strCategory: 'Beef',
        idCategory: '88',
        strCategoryThumb: 'beef',
        strCategoryDescription: 'Culinary name...',
    },
    {
        strCategory: 'Lamb',
        idCategory: '92',
        strCategoryThumb: 'lamb',
        strCategoryDescription: 'Culinary name...',
    }
];

const handleClick = () => (
    console.log('Test MapCategories and logging category name')
)

it('renders correctly', () => {
  const tree = renderer
    .create(<BrowserRouter><MapCategories categories={categories} handleClick={handleClick} /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});