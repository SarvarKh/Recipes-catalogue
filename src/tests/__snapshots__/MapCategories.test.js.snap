// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <button
    className="cat-item"
    onClick={[Function]}
    type="button"
  >
    <div
      className="category"
    >
      <img
        alt="Category"
        src="beef"
      />
      <div
        className="cat-name tooltip"
      >
        Beef
        <span
          className="category-desc tooltiptext"
        >
          Culinary name...
        </span>
      </div>
    </div>
  </button>,
  <button
    className="cat-item"
    onClick={[Function]}
    type="button"
  >
    <div
      className="category"
    >
      <img
        alt="Category"
        src="lamb"
      />
      <div
        className="cat-name tooltip"
      >
        Lamb
        <span
          className="category-desc tooltiptext"
        >
          Culinary name...
        </span>
      </div>
    </div>
  </button>,
]
`;
